src/cargo/ops/cargo_clean.rs

```diff
@@ -297,7 +297,8 @@ fn rm_rf(path: &Path, config: &Config, progress: &mut dyn CleaningProgressBar) -
         let entry = entry?;
         progress.on_clean()?;
         if entry.file_type().is_dir() {
-            paths::remove_dir(entry.path()).with_context(|| "could not remove build directory")?;
+            paths::remove_dir_all(entry.path())
+                .with_context(|| "could not remove build directory")?;
```

Comment:
The intention for why `remove_dir_all` is used here isn't very clear and someone could revert this in the future as "overkill".

While a comment could do it, I might recommend mirroring the logic in `paths::remove_dir_all` and keep `paths::remove_dir` and have the call for that fallback to `fs::remove_dir_all`.

Reply:
I added a comment to clarify this. `remove_dir` semantics are that the directory is *not* removed if it is not empty, which is not what we're actually trying to achieve here.

Reply:
Except the contents *should* have been deleted by this point, its just a race condition.

Reply:
Added explanation of race condition in comment